linode_api4/objects/firewall.py

```diff
@@ -0,0 +1,71 @@
+from linode_api4.errors import UnexpectedResponseError
+from linode_api4.objects import Base, DerivedBase, Property
+
+
+class FirewallDevice(DerivedBase):
+    api_endpoint = '/networking/firewalls/{firewall_id}/devices/{id}'
+    derived_url_path = 'devices'
+    parent_id_name = 'firewall_id'
+
+    properties = {
+        'created': Property(filterable=True, is_datetime=True),
+        'updated': Property(filterable=True, is_datetime=True),
+        'entity': Property(),
+        'id': Property(identifier=True)
+    }
+
+
+class Firewall(Base):
+    """
+    .. note:: This endpoint is in beta. This will only function if base_url is set to `https://api.linode.com/v4beta`.
+    """
+
+    api_endpoint = "/networking/firewalls/{id}"
+
+    properties = {
+        'id': Property(identifier=True),
+        'label': Property(mutable=True, filterable=True),
+        'tags': Property(mutable=True, filterable=True),
+        'status': Property(mutable=True),
+        'created': Property(filterable=True, is_datetime=True),
+        'updated': Property(filterable=True, is_datetime=True),
+        'devices': Property(derived_class=FirewallDevice),
+    }
+
+    @property
+    def rules(self):
+        """
+        Returns the JSON rules for this Firewall
+        """
+        return self._client.get('{}/rules'.format(self.api_endpoint), model=self)
+
+    def update_rules(self, rules):
+        """
+        Sets the JSON rules for this Firewall
+        """
+        return self._client.put('{}/rules'.format(self.api_endpoint), model=self, data=rules)
```

Comment:
This doesn't really need to return anything: 

```suggestion
        self._client.put('{}/rules'.format(self.api_endpoint), model=self, data=rules)
```

It should, however, call `self.invalidate()` - otherwise the cached rules on the object will be stale.